extensions/azurePublish/src/components/util.tsx

```diff
@@ -0,0 +1,100 @@
+import { getTenants, getARMTokenForTenant } from '@bfc/extension-client';
+import jwtDecode from 'jwt-decode';
+import { SharedColors } from '@uifabric/fluent-theme';
+import { AzureResourceTypes } from '../types';
+
+export function decodeToken(token: string) {
+  try {
+    return jwtDecode<any>(token);
+  } catch (err) {
+    console.error('decode token error in ', err);
+    return null;
+  }
+}
+
+export function removePlaceholder(config:any){
+  try{
+    if(config){
+      let str = JSON.stringify(config);
+      str = str.replace(/<[^>]*>/g, '');
+      const newConfig = JSON.parse(str);
+      return newConfig;
+    } else {
+      return undefined;
+    }
+  }catch(e){
+    console.error(e);
+  }
+};
+
+export function getExistResources (config){
+  const result = [];
+  if(config){
+    // If name or hostname is configured, it means the webapp is already created.
+    if(config.hostname || config.name){
+      result.push(AzureResourceTypes.WEBAPP);
+    }
+    if(config.settings?.MicrosoftAppId){
+      result.push(AzureResourceTypes.BOT_REGISTRATION);
+      result.push(AzureResourceTypes.APP_REGISTRATION);
+    }
+    if(config.settings?.luis?.authoringKey){
+      result.push(AzureResourceTypes.LUIS_AUTHORING);
+    }
+    if(config.settings?.luis?.endpointKey){
+      result.push(AzureResourceTypes.LUIS_PREDICTION);
+    }
+    if(config.settings?.qna?.subscriptionKey){
+      result.push(AzureResourceTypes.QNA);
+    }
+    if(config.settings?.applicationInsights?.InstrumentationKey){
+      result.push(AzureResourceTypes.APPINSIGHTS);
+    }
+    if(config.settings?.cosmosDb?.authKey){
+      result.push(AzureResourceTypes.COSMOSDB);
+    }
+    if(config.settings?.blobStorage?.connectionString){
+      result.push(AzureResourceTypes.BLOBSTORAGE);
+    }
+    return result;
+  } else return [];
+}
+
+export const iconStyle = (required) => {

```

Comment:
Please move this to wherever it's used, I don't see it being used in any other place, so no need to be exported.

(change name to .ts after you moved it)